docs/testing/unit-testing.md

```diff
@@ -0,0 +1,406 @@
+
+## Prerequisites
+
+This article assumes that you are familiar with the basic concepts of unit testing and mocking, and that you have some experience with a unit testing framework.
+
+> [!TIP]
+> If you are working with Visual Studio, see the article [Unit testing JavaScript and TypeScript in Visual Studio](/visualstudio/javascript/unit-testing-javascript-with-visual-studio.md).
+
+## Install the tool
+
+To install the library open a command prompt, navigate to the root of your add-in project, and then enter the following command:
+
+```command&nbsp;line
+npm install office-addin-mock --save-dev
+```
+
+## Basic usage
+
+1. Require (or import) the library to your test file. The following code is an example:
+
+   ```javascript
+   const OfficeMockObject = require("office-addin-mock");
+   ```
+
+1. Require (or import) the module that contains the add-in function that you want to test. The following is an example that assumes your test file is in a subfolder the folder with your add-in's code files:
+
+   ```javascript
+   const myOfficeAddinFeature = require("../my-office-add-in");
+   ```
+
+1. Create a data object that has the properties and subproperties that you need to mock to test the function. The following is an example of an object that mocks the Excel [Workbook.range.address](/javascript/api/excel/excel.range#address) property and the [Workbook.getSelectedRange](/javascript/api/excel/excel.workbook#getSelectedRange__) method. This isn't the final mock object. Think of it as a seed object that is used by `OfficeMockObject` to create the final mock object.
+
+   ```javascript
+   const mockData = {
+     workbook: {
+       range: {
+         address: "C2:G3",
+       },
+       getSelectedRange: function () {
+         return this.range;
+       },
+     },
+   };
+   ```
+
+1. Pass the data object to the `OfficeMockObject` constructor. Note the following about the returned `OfficeMockObject` object:
+
+   - It is a simplified mock of an [OfficeExtension.ClientRequestContext](/javascript/api/office/officeextension.clientrequestcontext) object. (It is also possible to mock a host object, such as [Excel](/javascript/api/excel) or [Word](/javascript/api/word). For an example, see [Mocking a host object](#mocking-a-host-object).)
+   - The mock object has all the members of the data object and also has mock `load()` and `sync()` methods.
+   - The mock object will mimic crucial error behavior of the `ClientRequestContext` object. For example, if the Office API you are testing tries to read a property without first loading the property and calling `sync()`, then the test will fail with an error similar to what would be thrown in production runtime: "Error, property not loaded".
+
+   ```javascript
+   const contextMock = new OfficeMockObject.OfficeMockObject(mockData);
```

Comment:
I didn't see this creation pattern in the linked docs. I saw `const wordMock = new officeAddinMock.OfficeMockObject(WordMockData);` and `const officeMock = new OfficeMockObject(PowerPointMockData);`. This confused me. Should that first example have a capital "O"?

Reply:
My bad. I forgot to add the following to the PR description. I'll add it now.

NOTE TO REVIEWERS: There will be cross-links between this article and the Readme of the repo for Office-Addin-Mock. So, the content of the two has to be coordinated and made consistent. There's no practical way to do that until after this article is close to finalized. Then there will be a separate PR for the the Readme. Nothing will be published until both documents are done and mutually consistent.